sklearn/datasets/_openml.py

```diff
@@ -680,6 +766,25 @@ def fetch_openml(
     delay : float, default=1.0
         Number of seconds between retries.
 
+    parser : {"auto", "liac-arff", "pandas"}, default="auto"
+        Parser used to load the ARFF file. `"pandas"` is the most efficient
+        parser but requires Pandas to be installed and is only used with
+        dense datasets. `"liac-arff"` is a pure Python ARFF parser that
+        is less efficient memory- and CPU-wise but it deals with sparse
+        datasets. `"auto"` selects `"pandas"` if Pandas is installed and that
+        the dataset to fetch is dense, otherwise `"liac-arff"`.
```

Comment:
Since the two parsers do not always return the same dtypes for integer columns, I would rather remove the "auto" option that depends on whether "pandas" is installed or not.

We already require `pandas` by default for `fetch_openml` because "`as_frame=True` by default, so I think we can set `parser="pandas"` by default instead.

Reply:
`as_frame="auto"` by default. We reverted it some version ago.